pkg/codegen/nodejs/gen.go

```diff
@@ -42,6 +42,11 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
 )
 
+// The minimum version of @pulumi/pulumi compatible with the generated SDK.
+const MinimumValidSDKVersion string = "^3.42.0"
+const MinumumTypescriptVersion string = "^4.3.5"
+const MinumumNodeTypesVersion string = "^18.11.9"
```

Comment:
We currently support running on Node.js 14+. Should we make this `^14` to limit the compiler to only the APIs available on v14?

```suggestion
const MinumumNodeTypesVersion string = "^14"
```

Looks like this is what we do for our program templates: 

https://github.com/pulumi/templates/blob/926b7531abcf8fd838bc6811118a1b1f2c958d95/typescript/package.json#L5

Reply:
I'd actually suggest `>=14` here. We don't want to conflict if the user wants to install `@types/node@^18`.